src/main/java/com/es/phoneshop/model/cart/HttpSessionCartService.java

```diff
@@ -22,15 +22,22 @@ private HttpSessionCartService() {
     }
 
     public static CartService getInstance() {
+        if(INSTANCE == null){
+            synchronized (HttpSessionCartService.class){
+                if(INSTANCE == null){
+                    INSTANCE = new HttpSessionCartService();
+                }
+            }
+        }
```

Comment:
Here and in other places - coding style guides:

if(INSTANCE == null){ --> if (INSTANCE == null) {

Style guides: https://google.github.io/styleguide/javaguide.html

Let IDEA format the file for you: https://www.jetbrains.com/help/idea/reformat-file-dialog.html

Ctrl + Shift + Alt + L or appropriate option in settings.